lua/lspconfig/typeprof.lua

```diff
@@ -0,0 +1,22 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = 'typeprof'
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'typeprof', '--lsp', '--stdio' },
+    filetypes = { 'ruby', 'eruby' },
+    root_dir = util.root_pattern('Gemfile', '.git'),
```

Comment:
do you want the search to search for `.git` at each step, or do you want to first search up the filesystem for gemfile and subsequently search for .git? Either way is fine. If it's the latter use find_git_ancestor.